fbgemm_gpu/fbgemm_gpu/split_table_batched_embeddings_ops.py

```diff
@@ -1942,8 +1942,8 @@ def _update_cache_miss_counter(
         lxu_cache_locations: Tensor,
         linear_cache_indices: Tensor,
     ) -> None:
-        CACHE_MISS = torch.tensor([-1], device=self.current_device, dtype=torch.int32)
-        CACHE_HIT = torch.tensor([-2], device=self.current_device, dtype=torch.int32)
+        CACHE_MISS = torch.tensor([-1], device=self.current_device, dtype=linear_cache_indices.dtype)
+        CACHE_HIT = torch.tensor([-2], device=self.current_device, dtype=linear_cache_indices.dtype)
```

Comment:
Qq: Why do we make this change?

Reply:
The `expected ... but found ...` error occurs in two tests in ROCm. I don't know if it is caused by any recent code change of the tests.

If one keeps `dtype=torch.int32`, the following error message is thrown when testing `test_nbit_cache_miss_counter`:
> RuntimeError: expected scalar type long int but found int

And if one changes to `dtype=torch.int64`, the `test_nbit_cache_update_function` complains
> RuntimeError: expected scalar type int but found long int

These errors are both thrown by https://github.com/pytorch/FBGEMM/blob/64a5c4a9f7f18cbf0e2afcaa561d96c949afd950/fbgemm_gpu/fbgemm_gpu/split_table_batched_embeddings_ops.py#L1949

Reply:
In general, applying a change that happens to work for some unknown reasons to already running code is not a good idea. I would highly appreciate it if you could investigate why it happens only for ROCm/AMD GPUs but not for CUDA/NVIDIA GPUs.

Reply:
> 

Thank you for your question. We investigated this issue. It is because of the old Pytorch version, and an issue closed recently (https://github.com/pytorch/pytorch/issues/9190). We adopted a newer version of Pytorch in the CI, and the issue was resolved.